.github/workflows/meson.yml

```diff
@@ -0,0 +1,104 @@
+name: Meson CI
+
+on:
+  push:
+    branches: [ master, ci ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {
+            name: Windows MSVC Release,
+            os: windows-latest,
+            msvc: true,
+            buildtype: release,
+            args: '-Ddefault_library=static --wrap-mode=forcefallback'
+          }
+          - {
+            name: Windows MinGW Release,
+            os: windows-latest,
+            msvc: false,
+            buildtype: release,
+            args: ''
+          }
+          - {
+            name: Ubuntu Debug,
+            os: ubuntu-latest,
+            buildtype: debugoptimized,
+            args: ''
+          }
+          - {
+            name: Ubuntu Release,
+            os: ubuntu-latest,
+            buildtype: release,
+            args: ''
+          }
+          - {
+            name: macOS Debug,
+            os: macos-latest,
+            buildtype: debugoptimized,
+            args: ''
+          }
+          - {
+            name: macOS Release,
+            os: macos-latest,
+            buildtype: release,
+            args: -Ddefault_library=static
+          }
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Setup Meson
+        run: |
+          python -m pip install --upgrade pip
+          pip install meson
+
+      - name: Setup MSVC
+        if: matrix.config.os == 'windows-latest' && matrix.config.msvc == true
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Install dependencies
+        shell: bash
+        run: |
+          case "${{ matrix.config.os }}" in
```

Comment:
I'm not a LibAss maintainer, but instead of using ``matrix.config.os``, you should probably use ``runner.os`` as the [docs](https://docs.github.com/en/actions/learn-github-actions/variables#detecting-the-operating-system) suggest.
To know the possibles values of ``runner.os``, see [this](https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context).

Reply:
the linked docs suggest using `runner.os` to be able to change hardcoded `runs-on` values. We set `runs-on` based on `matrix.config.os` here so there’s no difference wrt to runner types, but `matrix.config.os` also gives us more information about distro and version of the runner should we ever need it  
The existing CI file also uses containers in addition to GHA’s native envs. The selected container env can _only_ be obtained from `matrix` values since `runner.os` will just show the generic system type of the host runner

So in short, i think it’s fine as is and our CI files don’t fall into `runner.os` usecase